components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java

```diff
@@ -0,0 +1,768 @@
+
+        parameterValue.put(key, parameters);
+    }
+
+    public WriteResult insert() throws MongoQueryException {
+
+        if (!matchArguments(this.queryJson)) {
+            throw new MongoQueryException("Parameter count not matched with query parameters");
+        } else {
+            if (convertToDBObject(defaultQuery)) {
+                return this.collection.insert(this.query);
+            } else {
+                throw new MongoQueryException("Query format is invalid no collection found");
+            }
+        }
+    }
+
+    public DBCursor find() throws MongoQueryException {
+        if (!matchArguments(this.queryJson)) {
+            throw new MongoQueryException("Parameter count not matched with query parameters");
+        } else {
+            if (convertToDBObject(defaultQuery)) {
+
+                if (this.projection == null && this.query == null) {
+                    return this.collection.find();
+                } else if (this.projection == null) {
+                    return this.collection.find(this.query);
+                } else {
+                    return this.collection.find(this.query, this.projection);
+                }
+            } else {
+                throw new MongoQueryException("Query format is invalid no collection found");
+            }
+        }
+    }
+
+    public List distinct() throws MongoQueryException {
+        if (!matchArguments(this.queryJson)) {
+            throw new MongoQueryException("Parameter count not matched with query parameters");
+        } else {
+            if (convertToDBObject(defaultQuery)) {
+                return this.collection.distinct(this.distinctKey, this.query);
+            } else {
+                throw new MongoQueryException("Query format is invalid no collection found");
+            }
+        }
+    }
+
+    public void multiLookUp(boolean status) {
+        multipleLookUp = status;
+    }
+
+    public AggregationOutput aggregate() throws UserStoreException {
+        JSONObject defaultObject = new JSONObject(defaultQuery);
+        getAggregrationObjects(defaultObject);
+        try {
+            List<DBObject> pipeline = new ArrayList<DBObject>();
+            //add lookup attribute to pipeline
+            if (mapLookUp != null) {
+                if (isMultipleLookUp()) {
+                    DBObject lookup = new BasicDBObject("$lookup", new BasicDBObject(mapLookUp));
+                    pipeline.add(lookup);
+                } else {
+                    int track = 0;
+                    //add the json query object to aggregration pipeline in order manner
+                    while (track < multiMapLookup.size()) {
+                        for (Map<String, Object> map : multiMapLookup) {
+                            if (map.containsKey("dependency")) {
+                                map.remove("dependency");
+                                DBObject lookup = new BasicDBObject("$lookup", new BasicDBObject(map));

```

Comment:
You can use String constants if the Strings are getting duplicated in multiple places.